frontend/src/components/DicomViewer.tsx

```diff
@@ -0,0 +1,121 @@
+import { navigation } from '../utils/size_vars'
+
+
+function DicomViewer() {
+
+  // Initialize cornerstone only once, when component is rendered for the first time
+  React.useEffect(() => {
+      initCornerstone()
+  }, [])
+
+  // Set state variables
+  const [viewportIds, setViewportIds] = React.useState<number[]>([0])
+  const [activeViewportIndex, setActiveViewportIndex] = React.useState<number>(0)
+  // const [viewportData, setViewportData] = React.useState<string[]>(['', '', ''])
+  
+  // Set default viewport data for debugging
+  const viewportData = [
+    'https://marketing.webassets.siemens-healthineers.com/fcc5ee5afaaf9c51/b73cfcb2da62/Vida_Head.MR.Comp_DR-Gain_DR.1005.1.2021.04.27.14.20.13.818.14380335.dcm',
+    'https://marketing.webassets.siemens-healthineers.com/fcc5ee5afaaf9c51/b73cfcb2da62/Vida_Head.MR.Comp_DR-Gain_DR.1005.1.2021.04.27.14.20.13.818.14380335.dcm'
+  ]
+
+  return (
+      <Stack gap={1} sx={{ display: 'flex', flexDirection: 'column', flexGrow: 1, width: '100%', height: `calc(100vh - ${navigation.height})`, p: 1}}>
+      <Card sx={{ display: 'flex', flexDirection: 'row', p: 0.4, pl: 1 }}>
+        <Select
+          variant='plain'
+          indicator={<KeyboardArrowDown />}
+          startDecorator={<GridViewSharpIcon />}
+          defaultValue={[0]}
+          onChange={(event, ids) => {
+            ids ? setViewportIds(ids) : [0]
+          }}
+        >
+          <Option value={[0]}>1 view</Option>
+          <Option value={[0, 1]}>2 views</Option>
+          {/* <Option value={[0, 1, 2]}>3 views</Option> */}
+        </Select>
+        <Divider orientation='vertical' />
+        <DicomViewerToolbar />
+      </Card>
+
+      {/* Cornerstone viewports */}
+      <Grid container spacing={1} alignItems='stretch' direction='row' sx={{ height: '100%' }}>
+        {viewportIds.map((index) => (
+          <Grid xs={12 / viewportIds.length} key={index}>
+            <Card
+              variant={index === activeViewportIndex ? 'outlined' : 'plain'}
+              color={index === activeViewportIndex ? 'primary' : 'neutral'}
+              sx={{ p: 0.5, bgcolor: '#000000', height: '100%', border: '5px solid' }}
+            >
+              {
+                // The following condition checks, if some viewportData is defined for this index
+                // If an invalid imageId is set to the CornerstoneViewport, the component does not work
+                // properly. Thus the
+                viewportData[index] ? (
+                  <CornerstoneViewport
+                    key={index}
+                    imageIds={[`dicomweb:${viewportData[index]}`]}
+                    tools={tools}
+                    setViewportActive={() => {
+                      setActiveViewportIndex(index)
+                    }}
+
+                    // >> Save the following code snippets for later, might be useful
+                    // loadingIndicatorComponent={ CircularProgress }
```

Comment:
shall we start a separate snippet galery?

Reply:
no, the snipped is needed in that file, not somewhere else